internal/app/domain/auth.go

```diff
@@ -0,0 +1,53 @@
+package domain
+
+import (
+	"context"
+)
+
+
+type AuthClient interface {
+	// Login authenticates a user and returns a token.
+	Login(ctx context.Context, req *LoginRequest) (*LoginResponse, error)
+
+	// Register creates a new user account.
+	Register(ctx context.Context, req *RegisterRequest) (*RegisterResponse, error)
+
+	// VerifyToken validates a JWT token and returns its status.
+	VerifyToken(ctx context.Context, token string) (*VerifyTokenResponse, error)
+}
+
+// LoginRequest represents the request payload for the Login method.
+type LoginRequest struct {
+	Username string `json:"username"`
```

Comment:
These JSON struct tags should be in models defined for the REST API layer so duplicate this into something like auth_models.go in the handlers module in sticking with layer seperation